### PR TITLE
Critical Error in Checkout process - Context.php

### PR DESCRIPTION
**Pull request by vavrecan**
bug in customer update

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | all
| Description?  | bug in checkout process
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  
| How to test?  | do checkout with guest user, add two addresses. go to step 3 delivery. return to step 1 and continue with same account. It will throw error at step 3: Unfortunately, there are no carriers available for your delivery address.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8619)

### CODE REVIEW

**Comment by vavrecan** [comment]
ok to shortly put it: 

do checkout with guest user, add two addresses. go to step 3 delivery. return to step 1 and continue with same account. It will throw error at step 3: 
Unfortunately, there are no carriers available for your delivery address.

This is because the updateCustomer was called and it did not updated ps_cart_product delivery addresses for each item which is needed to get correct carriers.

Cheers.

**Comment by vavrecan** [comment]
guys would be really nice if you would return tab Issues in github. I understand that you hide it because people reported security issues but you know :)

**Comment by vavrecan** [comment]
added solution - call to cart->updateAddressId  which handles the id delivery address update  correctly.

**Comment by mickaelandrieu** [comment]
Hello @vavrecan , can you please complete the PR description as asked by Preston?

Thank you

**Comment by vavrecan** [comment]
sorry i can not override it. It throws error when i try to save.

**Comment by vavrecan** [comment]
please update the PR. I recorded the video reproducing the issue https://www.youtube.com/watch?v=hVwYa0hbmIs&feature=youtu.be

**Comment by ghost** [comment]
git-edit

**Comment by vavrecan** [comment]
bug in customer update



| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | all
| Description?  | bug in checkout process
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  
| How to test?  | do checkout with guest user, add two addresses. go to step 3 delivery. return to step 1 and continue with same account. It will throw error at step 3:
Unfortunately, there are no carriers available for your delivery address.

**Comment by vavrecan** [comment]
if i try to change main comment it throws error:  The content you are editing has changed. Please try again.

please go on and update it

**Comment by mickaelandrieu** [comment]
@vavrecan you need to do your pull request on `develop` branch, we don't use `master`. To summarize:
* `develop` => our "master" branch, => the future `1.7.4.0` at this moment
* `1.7.x.y` => 1.7 branch, our current major version
* `1.6.1.x` => 1.6 branch, the legacy version of PrestaShop maintained until august of 2018

So, if you want to fix this bug in `1.6`, create a pull request on `1.6.1.x` branch, if you want to fix this bug it's still time to create a pull request on `1.7.3.x` branch.

Thanks for your understanding

**Comment by vavrecan** [comment]
To be honest, I only reported it here since you closed issues on github. I will not put any more commits but if anybody will encounter same issue, I will help. You can reproduce and make patch now in any branch you like. The workflow Is too chaotic now and reporting single issue takes forever and its very hard to follow your guidelines. Take inspiration from other open source projects. Thank you for understanding.

**Comment by vavrecan** [comment]
The bug is present in all PS versions at the moment.

**Comment by mickaelandrieu** [comment]
> Take inspiration from other open source projects. Thank you for understanding.

I do it right now: closed.

Thanks for your help!
